src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,169 @@
+import {Feature, Settings} from '../../utils/settings'
+import {Navigation} from '../../roam/navigation'
+import {browser} from 'webextension-polyfill-ts'
+import {createPopper, Instance} from '@popperjs/core'
+
+export const config: Feature = {
+    id: 'live_preview',
+    name: 'Live Preview',
+    defaultIsActive: false,
+    warningTooltip: 'Experimental feature; Large databases might see performance issues.',
+    settings: [],
+}
+
+Settings.isActive('live_preview').then(active => {
+    if (active) {
+        enableLivePreview()
+    }
+})
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message?.featureId === 'live_preview' && message?.type === 'toggle') {
+        if (message.value) {
+            enableLivePreview()
+        } else {
+            removeLivePreview()
+        }
+    }
+})
+
+const createPreviewIframe = () => {
+    const iframe = document.createElement('iframe')
+
+    const url = Navigation.getPageUrl('search')
+    const isAdded = (pageUrl: string) => !!document.querySelector(`[src="${pageUrl}"]`)
```

Comment:
I'm not quite sure what this function checks for, can you give it a more descriptive name? also it seems to be duplicated elsewhere

Reply:
yes, duplication can be removed. This checks if a particular URL is already added to the dom. It is needed because on every hover, our mouseover is getting triggered multiple times, so we might end up adding the iframe multiple times

Reply:
name it something similar to `hasUrlLoaded(url)` maybe ?